src/main/java/team9/ddang/global/service/S3Service.java

```diff
@@ -0,0 +1,101 @@
+package team9.ddang.global.service;
+
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.UUID;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class S3Service {
+
+    private final AmazonS3Client amazonS3Client;
+
+    @Value("${cloud.aws.s3.bucket}")
+    private String bucket;
+
+    // MultipartFile을 전달받아 File로 전환한 후 S3에 업로드
+    public String upload(MultipartFile multipartFile, String dirName) throws IOException {
+
+        if(multipartFile == null || multipartFile.isEmpty()){
+            return null;
+        }
+
+        File uploadFile = convert(multipartFile)
+                .orElseThrow(() -> new IllegalArgumentException("파일 변환 실패"));
+
+        return upload(uploadFile, dirName);
+    }
+
+    private String upload(File uploadFile, String dirName){
+
+        String fileName = setFileName(dirName);
+
+        String uploadImageUrl = putS3(uploadFile, fileName);
+
+        removeNewFile(uploadFile); // convert() 과정에서 로컬에 생성된 파일 삭제
+
+        return uploadImageUrl;
+    }
+
+    private String putS3(File uploadFile, String fileName){
+
+        amazonS3Client.putObject(
+                new PutObjectRequest(bucket, fileName, uploadFile) // PublicRead 권한으로 upload
+        );
```

Comment:
그 주석은 달아두셨는데, 없는 것 같아서 조심스럽게 제안해봅니다.

```suggestion
        amazonS3Client.putObject(
                new PutObjectRequest(bucket, fileName, uploadFile) // PublicRead 권한으로 upload
                                .withCannedAcl(CannedAccessControlList.PublicRead)
        );
```

Reply:
아하 넵 감사합니다!